kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/system/SystemResourceInfo.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eurotech and/or its affiliates and others
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Eurotech
+ ******************************************************************************/
+package org.eclipse.kura.system;
+
+public class SystemResourceInfo {
+
+    private String name;
+    private String version;
+    private SystemResourceType type;
+
+    public SystemResourceInfo(String name) {
```

Comment:
I am in doubt with this: wouldn't be better to have this constructor set something for the version and type?
Don't know maybe "Unknown" with the proper types for both? So that we don't have to deal with empty values or nulls.